Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -130,26 +133,50 @@ public class ApolloCodegen {
     }
   }
 
-  /// Performs validation against deterministic errors that will cause code generation to fail.
-  static func validate(config: ConfigurationContext) throws {
-    if case .swiftPackage = config.output.testMocks,
-        config.output.schemaTypes.moduleType != .swiftPackageManager {
+  /// Validates the configuration against deterministic errors that will cause code generation to
+  /// fail. This validation step does not take into account schema and operation specific types, it
+  /// is only a static analysis of the configuration.
+  ///
+  /// - Parameter config: Code generation configuration settings.
+  public static func _validate(config: ApolloCodegenConfiguration) throws {
```

Comment:
The public method prefixed with `_` to indicate that while it's public it is still considered private.